mp2-v1/tests/common/cases/indexing.rs

```diff
@@ -570,17 +573,99 @@ impl TableIndexing {
         Ok((
             Self {
                 value_column,
-                contract_extraction: ContractExtractionArgs {
+                contract_extraction: Some(ContractExtractionArgs {
                     slot: StorageSlot::Simple(CONTRACT_SLOT),
-                },
-                contract,
+                }),
+                contract: Some(contract),
                 source,
                 table,
             },
             table_row_updates,
         ))
     }
 
+    pub(crate) async fn off_chain_test_case(
+        ctx: &mut TestContext,
+    ) -> Result<(Self, Vec<TableRowUpdate<BlockPrimaryIndex>>)> {
+        // build a table with the following columns:
+        // - "total_supply"
+        // - "conversion_rate"
+        // - "owner"
+        // - "token_id"
+        // where "owner" is the secondary index column and the pair of
+        // columns "(owner, token_id)" can be a primary key, that is
+        // they are sufficient to uniquely identify a row
+        const TABLE_NAME: &str = "Token Collection";
+        const SECONDARY_INDEX_COLUMN: &str = "owner";
+        const TOKEN_ID_COLUMN: &str = "token_id";
+        const REMAINING_COLUMN_NAMES: [&str; 2] = ["total_supply", "conversion_rate"];
+        let secondary_index_column = ColumnMetadata::PrimaryKey(SECONDARY_INDEX_COLUMN.to_string());
+        let non_indexed_columns = [
+            ColumnMetadata::PrimaryKey(TOKEN_ID_COLUMN.to_string()),
+            ColumnMetadata::NoPrimaryKey(REMAINING_COLUMN_NAMES[0].to_string()),
+            ColumnMetadata::NoPrimaryKey(REMAINING_COLUMN_NAMES[1].to_string()),
+        ];
+        let mut off_chain_data_args = OffChainDataArgs::new(
```

Comment:
Could we name it `OffchainTable` ?

Reply:
Renamed to `OffChainTableArgs` in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/415/commits/4a513aa8feffef4d226783367d061a59d56ecdd8 to keep naming consistent with the other table types